cmd/ocsp-responder/main.go

```diff
@@ -217,17 +212,23 @@ as generated by Boulder's single-ocsp command.
 	go cmd.DebugServer(c.OCSPResponder.DebugAddr)
 	go cmd.ProfileCmd(scope)
 
-	hd := &httpdown.HTTP{
-		StopTimeout: stopTimeout,
-		KillTimeout: killTimeout,
-	}
-	hdSrv, err := hd.ListenAndServe(srv)
-	cmd.FailOnError(err, "Error starting HTTP server")
+	done := make(chan bool)
+	go cmd.CatchSignals(logger, func() {
+		ctx, cancel := context.WithTimeout(context.Background(),
+			c.OCSPResponder.ShutdownStopTimeout.Duration)
+		defer cancel()
+		_ = srv.Shutdown(ctx)
+		done <- true
+	})
 
-	go cmd.CatchSignals(logger, func() { _ = hdSrv.Stop() })
+	err = srv.ListenAndServe()
+	if err != nil && err != http.ErrServerClosed {
+		cmd.FailOnError(err, "Running HTTP server")
+	}
 
-	forever := make(chan struct{}, 1)
-	<-forever
+	// Wait until done is signalled because ListenAndServer returns immediately on
+	// Shutdown, so we need to wait for Shutdown to return.
```

Comment:
I'm not sure I understand this comment. You say `ListenAndServer` (typo, should be `ListenAndServe`) returns immediately on `Shutdown` but also say we need to wait for `Shutdown` to return. Can you explain further?